backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -450,6 +454,20 @@ boolean childContains(Set<?> list, Set<?> sublist) {
     }
   }
 
+  private Map<String, Set<BlobExpanded>> getCollectToMap(
+      Map<String, Entry<BlobExpanded, String>> locationBlobsMapCommit) {
+    return locationBlobsMapCommit.values().stream()
+        .collect(
+            Collectors.toMap(
+                Entry::getValue,
+                entry -> new HashSet<>(Collections.singletonList(entry.getKey())),
+                (m1, m2) -> {
+                  HashSet<BlobExpanded> newHash = new HashSet<>(m1);
```

Comment:
Instead of HashSet please use the LinkedHashSet because we need to maintain the sequence order of record.

Reply:
fixed